lib/modules/dosomething/dosomething_campaign/dosomething_campaign.theme.inc

```diff
@@ -204,10 +204,19 @@ function dosomething_campaign_preprocess_action_page(&$vars, &$wrapper) {
   // Plan.
 
   // 1) Materials
+
+  // Build the share link here in case registering organ donors
+  $campaign_path = url(current_path(), array('absolute' => TRUE));
+  $custom_organ_share_link = $campaign_path . '?source=user/' . $vars['user']->uid;
```

Comment:
you can pass the `?source=user` in as an `$options` array with the [url](https://api.drupal.org/api/drupal/includes%21common.inc/function/url/7.x) function